Cargo.toml

```diff
@@ -28,15 +28,19 @@ bindgen = "0.69.4"
 [features]
 std = []
 plaintext-before-extension = []
+serde_support = ["serde", "std", "arrayvec/serde"]
 
 [profile.release]
 opt-level = 'z'   # Optimize for size
 lto = true        # Enable Link Time Optimization
 codegen-units = 1 # Reduce codegen units to improve optimizations
 
 [dependencies]
+serde = { version = "1.0", features = ["derive"], optional = true }
 bitflags = "2.4.2"
-arrayvec = "0.7.4"
+arrayvec = { version = "0.7.4", features = [
+    "serde",
+], optional = true, default-features = false }
```

Comment:
Is arrayvec really optional? You seem to use it without a feature gate in the code.
I think you just want
`arrayvec = { version = "0.7.4", default-features = false }`

Reply:
The `arrayvec/serde`  feature flag will enable the serde feature on arrayvec when necessary